src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c

```diff
@@ -885,6 +885,10 @@ int xocl_cleanup_mem(struct xocl_drm *drm_p)
 	int ret;
 	mutex_lock(&drm_p->mm_lock);
 	ret = xocl_cleanup_mem_nolock(drm_p);
+	if (drm_p->bo_usage_stat) {
+		vfree(drm_p->bo_usage_stat);
+		drm_p->bo_usage_stat = NULL;
+	}
```

Comment:
why these cleanup are not done inside xocl_cleanup_mem_nolock() while others are done there?

Reply:
Updated. Was not clear if that will work for xclbin with no DDR